qiskit_metal/renderers/renderer_gmsh/gmsh_renderer.py

```diff
@@ -0,0 +1,1129 @@
+            dz = sum(vac_height)
+            self.vacuum_box = gmsh.model.occ.addBox(x, y, z, dx, dy, dz)
+
+    def render_layer(self, layer_number: int, datatype: int = 0):
+        """Render the given layer number and datatype.
+
+        Args:
+            layer_number (int): number of the layer to render
+            datatype (int): number of the datatype. Defaults to 0.
+
+        Raises:
+            ValueError: if the required properties are not found
+                            in the layer-stack
+        """
+        props = ["thickness", "z_coord"]
+        result = self.parse_units_gmsh(
+            self.design.ls.get_properties_for_layer_datatype(
+                properties=props, layer_number=layer_number, datatype=datatype))
+        if result:
+            thickness, z_coord = result
+        else:
+            raise ValueError(
+                f"Could not find {props} for the layer_number={layer_number} and dataype="
+                f"{datatype}. Check your design and try again.")
+
+        chip_x = self.cc_x[layer_number] - self.cw_x[layer_number] / 2
+        chip_y = self.cc_y[layer_number] - self.cw_y[layer_number] / 2
+        chip_wx, chip_wy = self.cw_x[layer_number], self.cw_y[layer_number]
+
+        layer_tag = gmsh.model.occ.addBox(chip_x, chip_y, z_coord, chip_wx,
+                                          chip_wy, thickness)
+
+        if layer_number not in self.layers_dict:
+            self.layers_dict[layer_number] = -1
+
+        self.layers_dict[layer_number] = [layer_tag]
+
+    def subtract_from_layers(self):
+        """Subtract the QGeometries in tables from the chip ground plane"""
+        dim = 3
+        for layer_num, shapes in self.layer_subtract_dict.items():
+            shape_dim_tags = [(dim, s) for s in shapes]
+            layer_dim_tag = (3, self.layers_dict[layer_num][0])
+            subtract_layer = gmsh.model.occ.cut([layer_dim_tag], shape_dim_tags)
+            self.layers_dict[layer_num] = [subtract_layer[0][0][1]]
+
+    def fragment_interfaces(self):
+        """Fragment Gmsh surfaces to ensure consistent tetrahedral meshing
+        across interfaces between different materials.
+        """
+        all_vol_ids = list()
+        all_layer_geoms = defaultdict(dict)
+        all_dicts = (self.polys_dict, self.paths_dict)
+        for d in all_dicts:
+            for layer, geoms in d.items():
+                if layer not in all_layer_geoms:
+                    all_layer_geoms[layer] = dict()
+                all_layer_geoms[layer].update(geoms)
+
+        for _, geoms in all_layer_geoms.items():
+            for _, vol in geoms.items():
+                all_vol_ids += vol
+
+        for _, vol in self.layers_dict.items():
+            all_vol_ids += vol
+
+        vol_dimtags = [(3, vol) for vol in all_vol_ids]
+        gmsh.model.occ.fragment([(3, self.vacuum_box)], vol_dimtags)
+
+        # TODO: Do we require 3D junctions?
```

Comment:
Can you make an issue and add a reference to the issue in the code?

Reply:
Made issue #842